extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -139,38 +190,40 @@ TokenStateManager getTokenStateManager() {
         return tokenStateManager.get();
     }
 
-    private OidcTenantConfig getDynamicTenantConfig(RoutingContext context) {
-        OidcTenantConfig oidcConfig = null;
+    private Uni<OidcTenantConfig> getDynamicTenantConfig(RoutingContext context) {
         if (tenantConfigResolver.isResolvable()) {
-            oidcConfig = context.get(CURRENT_DYNAMIC_TENANT_CONFIG);
-            if (oidcConfig == null && context.get(CURRENT_DYNAMIC_TENANT_CONFIG_NULL) == null) {
```

Comment:
Hi @stuartwdouglas I've looked at the code - I believe this null check is about preventing a repeated call to the custom resolver if it previously returned `null` (if they do then the resolution continues by checking the static/default config) and the custom test resolver enforcers it, so I'd say this check needs to stay - I'll try a bit later on, may be Mon morning as it is a bit hard to concentrate right now on Fri evening :-)

Reply:
@stuartwdouglas  Hmm, I've checkout out your branch, confirmed that the tests fail because `oidc-tenancy/.../CustomTenantConfigResolver` sees it has been called more than once but your code here already protects against it since the null config is turned into null Uni so the next time the dynamic config is resolved it should recognize that null item and yet the provider is called again.
I wonder, is it possible that there is a race condition here ? `DefaultTenantConfigResolver` is called many times during a single request processing - but I'm peering into the code and can't see any problems :-)